pkg/cmd/resource/namespace.go

```diff
@@ -0,0 +1,88 @@
+package resource
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stripe/stripe-cli/pkg/ansi"
+)
+
+//
+// Public types
+//
+
+// NamespaceCmd represents namespace commands. Namespace commands are top-level
+// commands that are simply containers for resource commands.
+type NamespaceCmd struct {
+	Cmd          *cobra.Command
+	Name         string
+	ResourceCmds map[string]*ResourceCmd
+}
+
+//
+// Public functions
+//
+
+// NewNamespaceCmd returns a new NamespaceCmd.
+func NewNamespaceCmd(rootCmd *cobra.Command, namespaceName string) *NamespaceCmd {
+	cmd := &cobra.Command{
+		Use:         namespaceName,
+		Annotations: make(map[string]string),
+	}
+	cmd.SetUsageTemplate(namespaceUsageTemplate())
+
+	// For non-namespaced resources, we create a namespace command with the
+	// empty string as its name so we can group the resource commands in its
+	// ResourceCmds map, but we don't actually add the Cobra command as a
+	// subcommand.
+	if namespaceName != "" {
+		rootCmd.AddCommand(cmd)
+		rootCmd.Annotations[namespaceName] = "namespace"
+	}
+
+	return &NamespaceCmd{
+		Cmd:          cmd,
+		Name:         namespaceName,
+		ResourceCmds: make(map[string]*ResourceCmd),
+	}
+}
+
+//
+// Private functions
+//
+
+func namespaceUsageTemplate() string {
+	return fmt.Sprintf(`%s{{if .Runnable}}
```

Comment:
Lots of stuff in here. I assume that the new template is modeled after some default that you found inside Cobra?

Reply:
Yes. We're already overriding the default template at the root level here: https://github.com/stripe/stripe-cli/blob/b489043a11b42ffae12a3a98dd01e122d89ce393/pkg/cmd/root.go#L53-L96

in order to use ANSI bold sequences for the section names. Here I'm reusing the same pattern and also changing some section names ("commands" to "resources").

There might be a better, less repetitive way to do this though.